src/Compilers/Core/Portable/PEWriter/PdbWriter.cs

```diff
@@ -54,15 +55,52 @@ internal sealed class PdbWriter : IDisposable
         private uint[] _sequencePointEndLines;
         private uint[] _sequencePointEndColumns;
 
-        public PdbWriter(Stream stream, string fileName, Func<object> symWriterFactory = null)
+        [DllImport("shlwapi.dll", ExactSpelling = true, CharSet = CharSet.Unicode)]
+        private extern static IStream SHCreateMemStream([In] IntPtr pInit, [In] uint cbInit);
+
+        public PdbWriter(string fileName, Func<object> symWriterFactory = null)
         {
-            Debug.Assert(stream != null);
-            _stream = stream;
             _fileName = fileName;
             _symWriterFactory = symWriterFactory;
             CreateSequencePointBuffers(capacity: 64);
         }
 
+        public void WriteTo(Stream stream)
+        {
+            Debug.Assert(_nativeStream != null);
+            Debug.Assert(_symWriter != null);
+
+            try
+            {
+                // SymWriter flushes data to the native stream on close:
+                _symWriter.Close();
+                _symWriter = null;
+
+                var statStg = default(STATSTG);
+                _nativeStream.Stat(out statStg, grfStatFlag: 3 /*STATFLAG_NONAME | STATFLAG_NOOPEN*/);
+                _nativeStream.Seek(0L, 0, IntPtr.Zero);
+
+                // Copy unmanagedStream contents to stream in chunks
+                var len = (int)statStg.cbSize;
+                stream.SetLength(stream.Position + len);
```

Comment:
This line is probably what's failing the "non-seekable stream" unit test. I put it in because, when writing to a large file, it helps to give a hint to Windows how big the file is going to be. In this case we know exactly how big it's going to be.